### PR TITLE
Update cfn nag

### PR DESCRIPTION
**Pull request by tmcelhattan**
Changes Implemented:

- Updated gem set and CFN_NAG version to latest version
- Fixed Rubocop errors using newer version
- Added gem `simplecov` for unit testing coverage reports
- Refactored custom InternetGateway CFN_NAG rule to work with latest version
- Added unit tests for new custom CFN_NAG rule
- Added directory structure for creating/adding new custom CFN_NAG rules and unit tests for them.
- When running CFN_NAG in Jenkins pipeline, now scan entire `provisioning/cloudformation/templates/` directory instead of a single template file.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ VPC_STACK_TEMPLATE=./provisioning/cloudformation/templates/workshop-vpc.yml
 SUBNET_CIDR=10.10.21.0/24
 VPC_CIDR=10.10.21.0/22
 OWASP_ZAP_VERSION=2.5.0
-IMAGE_ID=ami-97785bed  # latest amzn linux
+IMAGE_ID=ami-00eb20669e0990cb4  # latest amzn linux
```
Comment (ghost) [inline deploy.sh:13]: for any account/region specific stuff like this, be sure it's called out in the readme for how to bring this guy up again.

File: pipeline/lib/cfn_nag/lib/cfn-nag/custom_rules/InternetGatewayRule.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W99'
+  end
+
+	def audit_impl(cfn_model)
```
Comment (ghost) [inline pipeline/lib/cfn_nag/lib/cfn-nag/custom_rules/InternetGatewayRule.rb]: not a big deal, but well formatted code is typically easier to read

**Review by ghost — APPROVED** [review]
looks good!  just make sure anything you had to do special to get this running in a fresh account is visible from the readme.

File: pipeline/lib/cfn_nag/lib/cfn-nag/custom_rules/InternetGatewayRule.rb (hunk 2 above)
Comment (tmcelhattan) [reply pipeline/lib/cfn_nag/lib/cfn-nag/custom_rules/InternetGatewayRule.rb]: Ok this should be fixed now, not sure why the format was funky lookin there lol

File: deploy.sh (hunk 1 above)
Comment (tmcelhattan) [reply deploy.sh:13]: Added  to the `Create Workshop Environment` section in the README for this

**Pull request merged**